### PR TITLE
 FEATURE: Mute user when ignoring the same user

### PR DESCRIPTION
**Pull request by khalilovcmded**
## Why?

This is part of the [Ability to ignore a user feature](https://meta.discourse.org/t/ability-to-ignore-a-user/110254/8).

If we ignore a user, either from the preferences or user summary or API , we will mute the same user too.

## Constraints

This can yield a janky behaviour on the UI, but I have a plan to fix that in a subsequent PR.

### CODE REVIEW

File: spec/services/user_updater_spec.rb [hunk 1]
```diff
@@ -52,6 +52,9 @@
       expect(IgnoredUser.where(user_id: u2.id).count).to eq 2
       expect(IgnoredUser.where(user_id: u1.id).count).to eq 2
       expect(IgnoredUser.where(user_id: u3.id).count).to eq 0
+      expect(MutedUser.where(user_id: u2.id).count).to eq 2
```
Comment (ZogStriP) [inline spec/services/user_updater_spec.rb]: Can we check that we have the proper Muted/Ignored user ids instead of just the #?

Comment (khalilovcmded) [reply spec/services/user_updater_spec.rb]: I agree. Then I will also refactor the `MutedUser` specs, not so safe.

File: spec/services/user_updater_spec.rb [hunk 2]
```diff
@@ -19,9 +19,9 @@
       updater = UserUpdater.new(u3, u3)
       updater.update_muted_users("")
 
-      expect(MutedUser.where(user_id: u2.id).count).to eq 2
-      expect(MutedUser.where(user_id: u1.id).count).to eq 2
-      expect(MutedUser.where(user_id: u3.id).count).to eq 0
+      expect(MutedUser.where(user_id: u2.id).pluck(:muted_user_id)).to match_array([u3.id, u1.id])
+      expect(MutedUser.where(user_id: u1.id).pluck(:muted_user_id)).to match_array([u2.id, u3.id])
+      expect(MutedUser.where(user_id: u3.id)).to match_array([])
```
Comment (ZogStriP) [inline spec/services/user_updater_spec.rb]: It's fine to use count when it's 0 😉

File: spec/services/user_updater_spec.rb [hunk 3]
```diff
@@ -49,9 +49,12 @@
       updater = UserUpdater.new(u3, u3)
       updater.update_ignored_users("")
 
-      expect(IgnoredUser.where(user_id: u2.id).count).to eq 2
-      expect(IgnoredUser.where(user_id: u1.id).count).to eq 2
-      expect(IgnoredUser.where(user_id: u3.id).count).to eq 0
+      expect(IgnoredUser.where(user_id: u2.id).pluck(:ignored_user_id)).to match_array([u3.id, u1.id])
+      expect(IgnoredUser.where(user_id: u1.id).pluck(:ignored_user_id)).to match_array([u2.id, u3.id])
+      expect(IgnoredUser.where(user_id: u3.id)).to match_array([])
```
Comment (ZogStriP) [inline spec/services/user_updater_spec.rb]:
```suggestion
      expect(IgnoredUser.where(user_id: u3.id).count).to eq(0)
```

**Comment by ZogStriP** [comment]
We'll be updating the queries to ensure we check for `ignored_users` whenever we check for `muted_users` instead of mucking with the lists.

**Pull request closed without merging**